test/src/tests/generated/paginggroup/models.go

```diff
@@ -174,7 +174,7 @@ func (page *OdataProductResultPage) Next() error {
 
 // NotDone returns true if the page enumeration should be started or is not yet complete.
 func (page OdataProductResultPage) NotDone() bool {
-	return !page.opr.IsEmpty()
+	return page.opr.OdataNextLink != nil || !page.opr.IsEmpty()
```

Comment:
What about the case where the next link is the empty string?  This happens for some services and is how we check if we need to advance in track 2, see [here](https://github.com/Azure/autorest.go/blob/track2/test/autorest/generated/paginggroup/pagers.go#L51).  I also think you can remove the `IsEmpty()` check unless you can find a case where this is actually necessary.

Reply:
To double confirm, does this function `NotDone` suppose to return `true` when there is not more pages?

BTW, I have changed this according to the track 2 reference.

Reply:
And I noticed that the track 2 generator is purely written in typescript, and frankly I think it is harder to read without the handy templates...

Reply:
Hmm it looks like this has caused some of the tests in the `paginggroup` to fail (search for "FAIL: paging_test.go" in the raw log).  Can you please investigate?

Reply:
I added the `IsEmpty` check back... hope this helps.